pkg/vpcmodel/synthesisOutput.go

```diff
@@ -61,23 +62,63 @@ func getSynthesisSpec(groupedLines []*groupedConnLine) *spec.Spec {
 	externals := spec.SpecExternals{}
 	sortGroupedLines(groupedLines)
 
+	bidirectionalMap := makeBidirectionalMap(groupedLines)
+
 	for _, groupedLine := range groupedLines {
+		if groupedLine.CommonProperties.Conn.isEmpty() {
+			continue
+		}
 		srcName, srcType := handleNameAndType(groupedLine.Src, externals)
 		dstName, dstType := handleNameAndType(groupedLine.Dst, externals)
-		if groupedLine.CommonProperties.Conn.isEmpty() {
+		bidirectional, ok := bidirectionalMap[getBidirectionalMapKeyByConnLine(groupedLine, true)]
+
+		if !ok {
+			// it means that it is bidirectional but the conn line will be appended to the list with the other direction.
 			continue
 		}
-		// For now, ignoring responsiveness of TCP connection
+
 		connLines = append(connLines, spec.SpecRequiredConnectionsElem{
 			Src:              spec.Resource{Name: srcName, Type: srcType},
 			Dst:              spec.Resource{Name: dstName, Type: dstType},
-			AllowedProtocols: sortProtocolList(spec.ProtocolList(connection.ToJSON(groupedLine.CommonProperties.Conn.allConn)))})
+			AllowedProtocols: sortProtocolList(spec.ProtocolList(connection.ToJSON(groupedLine.CommonProperties.Conn.allConn))),
+			Bidirectional:    bidirectional})
 	}
 	s.Externals = externals
 	s.RequiredConnections = connLines
 	return &s
 }
 
+func getBidirectionalMapKeyByNames(firstName, secName, conn string) string {
+	return fmt.Sprintf("%s-%s-%s", firstName, secName, conn)
```

Comment:
Better use a character that cannot be part of an endpoint name, such as `_` or `~`.
See the name field [here](https://cloud.ibm.com/apidocs/vpc/latest#create-vpc) for the list of legal characters.